src/main/groovy/de/dkfz/roddy/execution/ApptainerCommandBuilder.groovy

```diff
@@ -0,0 +1,357 @@
+package de.dkfz.roddy.execution
+
+
+import com.google.common.base.Preconditions
+import de.dkfz.roddy.tools.EscapableString
+import groovy.transform.AutoClone
+import groovy.transform.CompileStatic
+
+import org.jetbrains.annotations.NotNull
+import org.jetbrains.annotations.Nullable
+
+import java.nio.file.Path
+import java.nio.file.Paths
+
+import static de.dkfz.roddy.tools.EscapableString.Shortcuts.*
+
+
+/** A bind specification for mounting a host directory into a container. */
+@CompileStatic
+class BindSpec implements Comparable<BindSpec> {
+
+    enum Mode {
+        RO("ro"),
+        RW("rw");
+
+        String stringValue
+
+        private Mode(String value) {
+            stringValue = value
+        }
+
+        String toString() { stringValue; }
+
+        static Mode from(String value) {
+            value.toUpperCase() as Mode
+        }
+
+    }
+
+    /** The path of the mount as seen from the host-system (outside the container). */
+    @NotNull final Path hostPath
+    /** The path of the mount as seen from inside the container. If null, the hostPath is used. */
+    final Path containerPath
+    /** The mode of the mount, e.g. read-only or read-write. */
+    @NotNull final Mode mode
+
+    BindSpec(@NotNull Path hostPath,
+             Path containerPath = null,
+             @NotNull Mode opts = Mode.RO) {
+        Preconditions.checkNotNull(hostPath)
+        this.hostPath = hostPath
+        if (containerPath == null) {
+            this.containerPath = hostPath
+        } else {
+            this.containerPath = containerPath
+        }
+        Preconditions.checkNotNull(opts)
+        this.mode = opts
+    }
+
+    String toBindOption() {
+        "$hostPath:$containerPath:$mode"
+    }
+
+    String toString() {
+        toBindOption()
+    }
+
+    @Override
+    int compareTo(BindSpec other) {
+        Integer result = this.hostPath.compareTo(other.hostPath)
+        if (result != 0) return result
+
+        result = this.mode.compareTo(other.mode)
+        if (result != 0) return result
+
+        result = this.containerPath.compareTo(other.containerPath)
```

Comment:
True. I left it like that, because this is a multi-return method, and the other cases are coded in the same way. So here I valued the pattern higher than the brevity.